commonmark/src/main/java/org/commonmark/parser/InlineParser.java

```diff
@@ -2,14 +2,17 @@
 
 import org.commonmark.node.Node;
 
+import java.util.List;
+
 /**
  * Parser for inline content (text, links, emphasized text, etc).
  */
 public interface InlineParser {
 
     /**
-     * @param input the content to parse as inline
+     * @param lines the content to parse as inline
      * @param node the node to append resulting nodes to (as children)
      */
-    void parse(String input, Node node);
+    // TODO: Should we use a better type here, one that will be able to include source positions?
+    void parse(List<CharSequence> lines, Node node);
```

Comment:
I'm happy with this PR now (for internal API, would still need some more cleanup to be public API), but I still want to spike what this type would look like with source positions, so that we can avoid breaking API twice.